src/pages/workspaces/workspace/useWorkspace.test.ts

```diff
@@ -243,44 +250,147 @@ describe('useActiveWorkspace', () => {
     const uninitializedGoogleWorkspace = _.clone(initializedGoogleWorkspace)
     uninitializedGoogleWorkspace.workspaceInitialized = false
 
-    const expectedFirstStorageDetails = _.merge(defaultGoogleBucketOptions, defaultAzureStorageOptions)
-    const expectedSecondStorageDetails = _.merge({
+    const expectedStillSyncingDetails = _.merge(defaultGoogleBucketOptions, defaultAzureStorageOptions)
+    const expectedAllSyncedDetails = _.merge({
       googleBucketLocation: bucketLocationResponse.location,
-      googleBucketType: bucketLocationResponse.locationType
+      googleBucketType: bucketLocationResponse.locationType,
+      fetchedGoogleBucketLocation: 'SUCCESS'
     }, defaultAzureStorageOptions)
 
     // Act
     const { result, waitForNextUpdate } = renderHook(() => useWorkspace('testNamespace', 'testName'))
     await waitForNextUpdate() // For the call to checkBucketReadAccess to execute
 
     // Assert
-    assertResult(result.current, uninitializedGoogleWorkspace, expectedFirstStorageDetails, false)
+    assertResult(result.current, uninitializedGoogleWorkspace, expectedStillSyncingDetails, false)
     expect(workspaceStore.set).toHaveBeenCalledWith(uninitializedGoogleWorkspace)
     expect(WorkspaceUtils.updateRecentlyViewedWorkspaces).toHaveBeenCalledWith(uninitializedGoogleWorkspace.workspace.workspaceId)
     expect(GoogleStorage.saToken).toHaveBeenCalled()
 
     // Arrange
-    // Return success for the next call to checkBucketReadAccess
+    // Return success for the next call to checkBucketReadAccess, but fail storageCostEstimate.
+    let partialSuccessMockAjax: DeepPartial<AjaxContract> = {
```

Comment:
Yes, this test is painfully long. But splitting it up into a test for each individual Ajax call does not seem attractive to me.

Reply:
@blakery I went ahead and split this into smaller tests in the last commit(though I couldn't stop myself from creating a couple helper methods!). I agree it's more readable and focused now.